lib/dry/configurable/setting.rb

```diff
@@ -61,7 +61,7 @@ def initialize(name, input: Undefined, default: Undefined, **options)
         @default = default
         @options = options
 
-        evaluate if input_defined?
+        evaluate if input_defined? && !options[:constructor].is_a?(Proc)
```

Comment:
This can be simplified using the truthy value

```suggestion
        evaluate if input_defined? && !options[:constructor]
```

I added a reviewing to also be a `Proc` just to explain better the intention, but can be removed easyly.